src/components/ImagesCarousel.js

```diff
@@ -101,34 +108,71 @@ export const ImagesCarousel = ({ data, navigation, refreshTimeKey, aspectRatio }
     return renderItem({ item: carouselData[0] });
   }
 
+  if (isPaused) {
+    return (
+      <View>
+        {renderItem({ item: carouselData[index] })}
+        {showSliderPauseButton && pauseButton(isPaused, setIsPaused, sizeSliderPauseButton)}
+      </View>
+    );
+  }
+
   return (
-    <Carousel
-      data={carouselData}
-      renderItem={renderItem}
-      sliderWidth={dimensions.width}
-      itemWidth={itemWidth}
-      inactiveSlideScale={1}
-      autoplay
-      loop
-      autoplayDelay={0}
-      autoplayInterval={4000}
-      containerCustomStyle={styles.center}
-    />
+    <View>
+      <Carousel
+        data={carouselData}
+        renderItem={renderItem}
+        sliderWidth={dimensions.width}
+        itemWidth={itemWidth}
+        inactiveSlideScale={1}
+        autoplay
+        loop
+        autoplayDelay={0}
+        autoplayInterval={4000}
+        containerCustomStyle={styles.center}
+        onScrollIndexChanged={(index) => setIndex(index)}
+      />
+
+      {showSliderPauseButton && pauseButton(isPaused, setIsPaused, sizeSliderPauseButton)}
+    </View>
```

Comment:
what about merging the two returns into one? it seems that the last line is the same for showing the button, so following should work:

```
<View>
  {isPaused ? renderItem({ item: carouselData[index] }) : (<Carousel
    data={carouselData}
    renderItem={renderItem}
    sliderWidth={dimensions.width}
    itemWidth={itemWidth}
    inactiveSlideScale={1}
    autoplay
    loop
    autoplayDelay={0}
    autoplayInterval={4000}
    containerCustomStyle={styles.center}
    onScrollIndexChanged={(index) => setIndex(index)}
  />)}

  {showSliderPauseButton && pauseButton(isPaused, setIsPaused, sizeSliderPauseButton)}
</View>
```

Reply:
https://github.com/smart-village-solutions/smart-village-app-app/commit/6117184332afa80c19596a82cb1e0cab2aa57624